### ISSUE TITLE
Control flow?

### ISSUE THREAD

**Issue by ericlemasson** [CoatiSoftware/SourcetrailBugTracker#576]
[New feature]
Any plans to add in the future automatic creation of visual control flow for any method / function, similar to what understand is doing?

**Comment by ericlemasson** [comment]
yes, I like it for average size function, when it is still readable and useful (especially to diff with previous version). But I know this would be complex to develop.

**Comment by egraether** [comment]
Yes, it would be complex to develop and also require a lot more data to collect and store.

**Comment by ericlemasson** [comment]
I was thinking more of a 'on-demand functionality' ... when inside a function, could do  - right-click / generate control flow for the current function - to switch to a graphical view. I don't think cross-references are required here because we focus only on the current function. Clicking on an item in the graph would bring us back to the corresponding line of code in the function.
Not sure if this make any sense ...

**Comment by TomSmartBishop** [comment]
I agree that this would be a very useful functionality!

**Comment by egraether** [comment]
+1 by mail

**Comment by fabric-and-ink** [comment]
This would be extremely helpful. I think that LLVM is able to provide a CFG for any function.

[Here](https://eli.thegreenplace.net/2013/09/16/analyzing-function-cfgs-with-llvm) is an old blog post about this idea. The API has probably changed in the meantime...
